homeassistant/components/camera/neato.py

```diff
@@ -0,0 +1,98 @@
+"""
+Camera that loads a picture from a local file.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.neato/
+"""
+import errno
+import logging
+import os
+
+from homeassistant.components.camera import Camera
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.components.neato import (
+    NEATO_MAP_DATA, NEATO_ROBOTS, NEATO_LOGIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Camera."""
+    if not hass.data[NEATO_ROBOTS]:
+        return False
+    directory_path = hass.config.config_dir
+    if not os.access(directory_path, os.R_OK):
+        _LOGGER.error("file path %s is not readable", directory_path)
+        return False
+    dev = []
+    for robot in hass.data[NEATO_ROBOTS]:
+        if 'maps' not in robot.traits:
+            return False
```

Comment:
Returning means that if you have 2 robots and 1 doesn't have a map, you get none. You probably mean `continue`. (or, since it is so small, you can just make it `if 'maps' in robot.traits: dev.append(…)`